foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,772 @@
+""" Mesh module
+
+This module contains the implementation to store, access,
+and modify a mesh and related data
+
+"""
+import uuid
+from simphony.cuds.abstractmesh import ABCMesh
+from simphony.core.cuba import CUBA
+from simphony.cuds.mesh import Point, Face, Cell
+import simphony.core.data_container as dc
+from foam_files import FoamFiles
+import simphonyfoaminterface as foamface
+from foam_templates import head
+from foam_templates import scalarTemplates, vectorTemplates
+import tempfile
+import os
+import numpy
+
+
+class FoamMesh(ABCMesh):
+    """ Proxy class to communicate with OpenFoam engines mesh data.
+
+    Parameters
+    ----------
+    name : str
+        name of mesh
+
+    mesh : ABCMesh
+       mesh to store
+
+    Attributes
+    ----------
+    name : str
+        name of mesh
+    data : DataContainer
+        DataContainer to store mesh global data (midpoint, ...).
+    _uuidToFoamLabel : dictionary
+        Mapping from uuid to OpenFoam label number
+    _foamCellLabelToUuid : dictionary
+        Mapping from OpenFoam cell label number to uuid
+    _foamFaceLabelToUuid : dictionary
+        Mapping from  OpenFoam face label number to uuid
+    _foamEdgeLabelToUuid : dictionary
+        Mapping from  OpenFoam edge label number to uuid
+    _foamPointLabelToUuid : dictionary
+        Mapping from  OpenFoam point label number to uuid
+
+    """
+
+    def __init__(self, name, mesh=None, path=None):
+        self.name = name
+        self.data = dc.DataContainer()
+        self.time = str(0)
```

Comment:
I'll make modifications so that time is set in the end of run -method and  that underscore marking is used in FoamMesh for time